### ISSUE TITLE
[PRE REVIEW]: Magphi: Sequence extraction tool from FASTA and GFF3 files using seed pairs

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#4007]
**Submitting author:** @milnus (<a href="http://orcid.org/0000-0001-9751-9877">Magnus Jespersen</a>)
**Repository:** https://github.com/milnus/Magphi
**Branch with paper.md** (empty if default branch): 
**Version:** v0.1.6
**Editor:** @majensen
**Reviewers:** @gtonkinhill, @dcnickle
**Managing EiC:** Kyle Niemeyer

 

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

## Status

[![status](https://joss.theoj.org/papers/a7fc4d9bcdf800adfd4dad43d244d684/status.svg)](https://joss.theoj.org/papers/a7fc4d9bcdf800adfd4dad43d244d684)

Status badge code:

```
HTML: <a href="https://joss.theoj.org/papers/a7fc4d9bcdf800adfd4dad43d244d684"><img src="https://joss.theoj.org/papers/a7fc4d9bcdf800adfd4dad43d244d684/status.svg"></a>
Markdown: [![status](https://joss.theoj.org/papers/a7fc4d9bcdf800adfd4dad43d244d684/status.svg)](https://joss.theoj.org/papers/a7fc4d9bcdf800adfd4dad43d244d684)
```

**Author instructions**

Thanks for submitting your paper to JOSS @milnus. **Currently, there isn't an JOSS editor assigned** to your paper.

The author's suggestion for the handling editor is @will-rowe.

@milnus if you have any suggestions for potential reviewers then please mention them here in this thread (without tagging them with an @). In addition, [this list of people](https://bit.ly/joss-reviewers) have already agreed to review for JOSS and may be suitable for this submission (please start at the bottom of the list).

**Editor instructions**

The JOSS submission bot @whedon is here to help you find and assign reviewers and start the main review. To find out what @whedon can do for you type:

```
@whedon commands
```

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks.

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by whedon** [comment]
Wordcount for `paper.md` is 493

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1093/bioinformatics/btu153 is OK
- 10.1101/pdb.top17 is OK
- 10.1371/journal.pone.0163962 is OK
- 10.1093/bioinformatics/btx610 is OK
- 10.1093/bioinformatics/btq033 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
```
Software report (experimental):

github.com/AlDanial/cloc v 1.88  T=0.26 s (124.5 files/s, 18456.9 lines/s)
-------------------------------------------------------------------------------
Language                     files          blank        comment           code
-------------------------------------------------------------------------------
Python                          13            767            735           2351
Bourne Shell                     3             55             69            269
Markdown                         2             32              0            142
XML                              1              1              0            117
YAML                             5              3             12             96
TeX                              1              5              0             66
Dockerfile                       1              4              4             10
JSON                             6              0              0              6
-------------------------------------------------------------------------------
SUM:                            32            867            820           3057
-------------------------------------------------------------------------------


Statistical information for the repository '1ca069326842b4f7f2ba78eb' was
gathered on 2021/12/17.
The following historical commit information, by author, was found:

Author                     Commits    Insertions      Deletions    % of changes
Andrew Hayes                     2            43              1            0.37
Magnus G. Jespersen              1             1              1            0.02
milnus                         129          7452           4540           99.62

Below are the number of rows from each author that have survived and are still
intact in the current revision:

Author                     Rows      Stability          Age       % in comments
Andrew Hayes                 43          100.0          0.5               32.56
milnus                     3810           51.1          1.7               10.89
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.04007/joss.04007/10.21105.joss.04007.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.04007/joss.04007/10.21105.joss.04007.pdf) :page_facing_up: :point_left:

**Comment by kyleniemeyer** [comment]
@whedon invite @will-rowe as editor

Hi @will-rowe, could you edit this one?

**Comment by whedon** [comment]
@will-rowe has been invited to edit this submission.

**Comment by kthyng** [comment]
@will-rowe Just another ping here in the new year — are you interested in editing this submission?

**Comment by danielskatz** [comment]
@whedon assign @will-rowe as editor

Sorry for the delay, @milnus. Will will take this on this weekend.

**Comment by whedon** [comment]
OK, the editor is @will-rowe

**Comment by milnus** [comment]
No worries about the delay.
Here are two reviewer suggestions, from the list of existing reviewers:
asafpr
ayaanhossain

**Comment by will-rowe** [comment]
Very sorry for the delay and thank you for the reviewer suggestions. Let's see if they can do it - @asafpr and @ayaanhossain , are you able to review this submission for JOSS?

**Comment by ayaanhossain** [comment]
Sure! Love to take a look.

**Comment by will-rowe** [comment]
Great - thanks @ayaanhossain! Not sure if you've reviewed for us before but what happens now is we find our second reviewer before starting the review process in a new issue tracker (but feel free to get started early if that works for you).

**Comment by will-rowe** [comment]
@whedon assign @ayaanhossain as reviewer

**Comment by whedon** [comment]
OK, @ayaanhossain is now a reviewer

**Comment by ayaanhossain** [comment]
@whedon commands

**Comment by whedon** [comment]
Here are some things you can ask me to do:

```
# List Whedon's capabilities
@whedon commands

# List of editor GitHub usernames
@whedon list editors

# List of reviewers together with programming language preferences and domain expertise
@whedon list reviewers

EDITORIAL TASKS

# Compile the paper
@whedon generate pdf

# Compile the paper from alternative branch
@whedon generate pdf from branch custom-branch-name

# Ask Whedon to check the references for missing DOIs
@whedon check references

# Ask Whedon to check repository statistics for the submitted software
@whedon check repository

```

**Comment by will-rowe** [comment]
Hi @shenwei356 - would you be interested in reviewing this JOSS submission?

**Comment by arfon** [comment]
:wave: @majensen – would you be willing to take this submission on instead of @will-rowe? Due to a change in professional circumstances Will is stepping down from the JOSS editorial team.

**Comment by majensen** [comment]
@editorialbot set me as editor

**Comment by majensen** [comment]
@editorialbot commands

**Comment by majensen** [comment]
@editorialbot assign me as editor

**Comment by majensen** [comment]
Hi @huddlej  -- would you have time and inclination to review [this work](#4007) for [JOSS](https://joss.theoj.org)? It would be great to have your microbe sequencing expertise here. Thanks for your consideration --@majensen

**Comment by majensen** [comment]
Hey @amoeba - have you got cycles to review [this work](#4007) for [JOSS](https://joss.theoj.org)? Consideration much appreciated - @majensen

**Comment by milnus** [comment]
Another reviewer suggestion from my side is: gtonkinhill
Not currently a reviewer with you, but has himself published genomic/bioinformatics tools

**Comment by majensen** [comment]
Hi @gtonkinhill - Would you be willing to review [this software submission](#4007) for our [Journal of Open Source Software (JOSS)](https://joss.theoj.org)? Please have a look at our [guide for reviewers](https://joss.readthedocs.io/en/latest/reviewer_guidelines.html), and let me know here in this thread, or via email to me at maj \-dot\- fortinbras \-at\- gmail \-dot\- com. Your expertise here would be invaluable. 
Really appreciate your consideration    --@majensen

**Comment by amoeba** [comment]
Hi @majensen, I don't have cycles for this at the moment, apologies.

**Comment by gtonkinhill** [comment]
Hi @majensen, I would be happy to review. 

I am currently visiting the Doherty Institute where @milnus is based but am employed at the University of Oslo. @milnus and I do not have any past or ongoing projects so I think I will be able to give an unbiased review but wanted to disclose the association just in case it is an issue.

**Comment by majensen** [comment]
@amoeba appreciate your quick response and consideration. Thanks!

**Comment by majensen** [comment]
@gtonkinhill thanks very much! I appreciate the transparency and I don't see any issues. 
(P.S., in olden times I used interact on occasion with your old boss Simon when I was working in HIV. Apropos of not much except what a small world this is...)

**Comment by majensen** [comment]
@editorialbot add @gtonkinhill as reviewer

**Comment by majensen** [comment]
@ayaanhossain just checking in, are you still on board to review this work? It has been some time and I want to confirm with you - thanks! --@majensen

**Comment by huddlej** [comment]
@majensen I'm sorry for not responding sooner, but I'm in a similar situation in that I don't have time for reviews at the moment. I'd be happy to help with reviews in the future, though.

**Comment by ayaanhossain** [comment]
Hi @majensen, unfortunately, I too am swamped with work and won't be able to commit to this reliably at the moment. I'd like to pass on for now but will definitely sign up for other reviews in the future.

**Comment by majensen** [comment]
Thanks @huddlej and @ayaanhossain - will catch you next time!

**Comment by majensen** [comment]
Hey @dcnickle my friend - would you have time to review [this work](#4007) for [JOSS](https://joss.theoj.org)? I really appreciate the consideration.

**Comment by dcnickle** [comment]
Yes - I can review.

David

**Comment by majensen** [comment]
@editorialbot remove @ayaanhossain as reviewer

**Comment by majensen** [comment]
@editorialbot add @dcnickle as reviewer

**Comment by majensen** [comment]
@dcnickle thanks man -

**Comment by majensen** [comment]
@editorialbot start review

**Comment by majensen** [comment]
@milnus @gtonkinhill @dcnickle the official review has begun in issue #4369 - we will continue the conversation there. Please let me know if you have any questions about the process - either in the thread or at maj \-dot\- fortinbras \-at\- gmail \-dot\- com. Thanks!

**Comment by dcnickle** [comment]
I need this to be assigned to @DavidNickle.

@dcnickle is broken

David

**Comment by majensen** [comment]
@editorialbot remove @dcnickle as reviewer
